pallets/staking/src/lib.rs

```diff
@@ -2445,18 +2447,20 @@ decl_module! {
             size: ElectionSize,
         ) -> DispatchResultWithPostInfo {
             ensure_none(origin)?;
-            Self::check_and_replace_solution(
+            let adjustments = Self::check_and_replace_solution(
                 winners,
                 compact,
                 ElectionCompute::Unsigned,
                 score,
                 era,
                 size,
-            )
-            // TODO: instead of returning an error, panic. This makes the entire produced block
-            // invalid.
-            // This ensures that block authors will not ever try and submit a solution which is not
-            // an improvement, since they will lose their authoring points/rewards.
+            ).expect(
+                "An unsigned solution can only be submitted by validators; A validator should \
+                always produce correct solutions, else this block should not be imported, thus \
+				effectively depriving the validators from their authoring reward. Hence, this panic
+				is expected."
```

Comment:
indentation

Reply:
It seems fine on my local, I don't know why it seems off over github

Reply:
Tab vs space issue. The bottom two lines are indented used tabs.